### DISCUSSION TITLE
[Question] Remote Cockpit OS and Packages Compability

### DISCUSSION THREAD

**Discussion by LazyGeniusMan** [cockpit-project/cockpit#17027] [Q&A] [answered]
If I install Cockpit on my server but not expose the GUI  because I want to access the GUI only via my local PC, does my server and local PC must have the exact same:
1. OS (ex: Debian, Ubuntu, etc)
2. OS Version (ex: focal, bullseye, etc)
3. Cockpit Packages Installed (cockpit, cockpit-ws, cockpit-storaged, etc)
4. Cockpit Packages Version

Or as long as both my server and local PC have cockpit installed it will work fine ? (I only update and add additional cockpit packages in my server, so that means I have outdated cockpit version and bare minimum cockpit packages installed in my local PC)

EDIT: oops I just realized the repo have discussion tab, should I move this question there?

**Comment by marusak** [comment]
If I understand correctly, you have server that you want to manage with Cockpit. And then you open browser on some other machine, but don't manage this machine with Cockpit.

Answer to that is - the other machine only needs web browser. You can use your phone, windows machine or whatever that has browser and internet connection (and is able to access your machine address).

If you also want manage your seconds machine with Cockpit it is still completely fine if they differ in any way.

> EDIT: oops I just realized the repo have discussion tab, should I move this question there?

moved

**Reply by LazyGeniusMan** [reply]
> Answer to that is - the other machine only needs web browser. You can use your phone, windows machine or whatever that has browser and internet connection (and is able to access your machine address).

I prefer not to open the port for cockpit web UI in my server, I just want to open standard port for SSH, HTTP, and HTTPS in my server. I will connect to cockpit web UI from my local PC

If I want to do that, my local PC also have to install cockpit right? so instead of go to `<Server IP>:9090`, I will go to `localhost:9090` and put my server IP in the login screen.

**Comment by martinpitt** [comment, marked as answer]
Note that your local PC/mobile/etc. does not need any cockpit installed, as long as the server has cockpit-ws.

The other supported variant is that the managed server does not have cockpit's web server (cockpit-ws), and you connect to it remotely, either from local cockpit ("add host" on the top left), or using the [Cockpit Client flatpak](https://github.com/flathub/org.cockpit_project.CockpitClient). That second variant is probably what you are after, as you said "but not expose the GUI on the server"? The only "expose" would be the open port from cockpit-ws.

**Reply by LazyGeniusMan** [reply]
> or using the [Cockpit Client flatpak](https://github.com/flathub/org.cockpit_project.CockpitClient).

This is neat, looks like it's only focusing managing remote machine, should be better than installing `cockpit` package in my local pc right ? or that flapak package do the same thing as cockpit package in official distro repo ? (Installation steps in cockpit docs)

Edit: oh it's a desktop app? I thought it's also web app. So If I want web app intead, then I install regular cockpit package ?

**Reply by martinpitt** [reply]
The flatpak is built and supported by us, it uses the same code -- just plumbed together differently to avoid opening port 9090 locally. The point is to be a desktop app, as the cockpit-ws package already provides the "web app". So, it's a matter of preference for you.

**Reply by LazyGeniusMan** [reply]
Ah I see, thanks
